starlette/responses.py

```diff
@@ -204,7 +204,7 @@ def __init__(
         media_type: str = None,
         background: BackgroundTask = None,
     ) -> None:
-        if inspect.isasyncgen(content):
+        if hasattr(content, '__aiter__') and hasattr(content, '__anext__'):
```

Comment:
This won't work in case when class has a custom `__aiter__` method but doesn't implement `__anext__`.

For instance:
```python
class NumbersStream:
    async def __aiter__(self):
        for i in range(100):
            yield i
```

Maybe it's better to change it to `isinstance(content, typing.AsyncIterable)`.
With this check, every object which is async iterable can be used as a streaming response.


```suggestion
        if isinstance(content, typing.AsyncIterable):
```